### ISSUE TITLE
VS Code API does not return `isSingleLine` true on triple-click line selection

### ISSUE THREAD

**Issue by smcenlly** [microsoft/vscode#114557]
Version: 1.52.1 (user setup)
Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
Date: 2020-12-16T16:34:46.910Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.19041

Steps to Reproduce:

1. Open a text document with some sample content, for example:
```text
This is my document
second line
third line
```
2. Outside of the bounds of line 1, triple click to select the line for `This is my document`.
3. The actual selection is line 1: col 0: to line 2: col 0.
4. Note that in the case of selecting the third line, the selection is line 3: col 0 to line 3: <eol>.

While the behavior is inconsistent, the issue I that when writing a VS Code extensions that uses the [TextEditorSelectionChangeEvent](https://code.visualstudio.com/api/references/vscode-api#TextEditorSelectionChangeEvent) event, the `isSingleLine` [Selection](https://code.visualstudio.com/api/references/vscode-api#Selection) property is not consistent and misses covering the case of triple-click selection. Instead, in my selection handling logic, I need to code similar to below to correctly handle this case:

```diff
- if (selection.isSingleLine) {
+ if ((selection.start.line === selection.end.line) || (selection.start.line === selection.end.line - 1 && selection.end.character === 0)) {
    // Do single-line selection behavior
 }
```

I don't mind the behavior but it would be ideal if the API correctly returned `isSingleLine` in this case.

Does this issue occur when all extensions are disabled?: Yes

**Comment by alexdima** [comment]
Triple click (or clicking on a line number) is implemented by setting the selection from the start of the current line to the start of the next line:
![image](https://user-images.githubusercontent.com/5047891/105177523-13dc7800-5b27-11eb-9440-85ca838d0006.png)

This means that the current text on the line plus its EOL sequence (`\n` or `\r\n`) is selected.

In our coordinate system, the position after `\n` or `\r\n` is not accounted on the current line, but on the next line. So basically the above screenshot represents a selection from `line: 21, char: 0` to `line: 22, char: 0`.

`Selection.isSingleLine` is speced to be `` "`true` if `start.line` and `end.line` are equal. ``. By now, we have very many extensions that depend on this, so I don't think we can ever fix this.

So, TL;DR. `Selection.isSingleLine` should probably return true for a selection from `21,0 -> 22,0` because that would make sense conceptionally, but the property is already specified to return `false` in this case, so we cannot change it without potentially breaking an unknown number of extensions.
